src/main/io/osd.c

```diff
@@ -400,6 +404,17 @@ static void osdFormatMessage(char *buff, size_t size, const char *message)
     if (message) {
         memcpy(buff, message, strlen(message));
     }
+
+    // Save warning into pilotConfig->warning, used for DJI OSD
+    // stored into another field for saving original pilot name
+    if (osdWarnGetState(OSD_WARNING_DJI)) {
+        osdConfigMutable()->item_pos[OSD_CRAFT_NAME] = osdConfig()->item_pos[OSD_WARNINGS]; // Change position of craft name
+        strncpy(pilotConfigMutable()->warning, " ", OSD_FORMAT_MESSAGE_BUFFER_SIZE); // Clear previous messages
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Easily used incorrectly; doesn't always \0-terminate or check for invalid pointers [MS-banned] (CWE-120).  Risk is low because the source is a constant character.](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5623733)